guillotina/api/service.py

```diff
@@ -36,6 +42,56 @@ def __setattr__(self, name, value):
 
 
 class Service(View):
+    __validator__ = None
+    _sentinal = object()
+
+    def _get_validator(self, config):
+        [schema, validator] = [None, None]
+        if(self.__validator__ is None and self.__validator__ != self._sentinal):
+            validator = self._sentinal
+            if config['requestBody']:
+                requestBody = config['requestBody']
+                if '$ref' in requestBody['content']['application/json']['schema']:
+                    try:
+                        ref = requestBody['content']['application/json']['schema']['$ref']
+                        schema_name = ref.split('/')[-1]
+                        schema = app_settings['json_schema_definitions'][schema_name]
+                        validator = get_schema_validator(schema_name)
+                    except KeyError:
+                        logger.warning('Invalid jsonschema', exc_info=True)
+                elif requestBody['content']['application/json']['schema'] is not None:
+                    try:
+                        schema = requestBody['content']['application/json']['schema']
+                        jsonschema_validator = jsonschema.validators.validator_for(schema)
+                        validator = jsonschema_validator(schema)
+                    except jsonschema.exceptions.ValidationError:
+                        logger.warning('Could not validate schema', exc_info=True)
+                else:
+                    logger.warning("No schema found in service definition")
+            else:
+                pass  # can be used for query, path or header parameters
+            return [schema, validator]
+        else:
+            return [self.__schema__, self.__validator__]
+
+    async def validate(self):
+
+        [self.__schema__, self.__validator__] = self._get_validator(self.__config__)
```

Comment:
Just change to `validator = self._get_validator(self.__config__)`

and then make next line be `if validator is not None:`